decaporg/decap-cms
# Fix: Set correct branch when it's not specified in the config

fix #5617 


**Summary**
Currently, the default branch in Github, Gitlab and Bitbucket is set to be `main` instead of `master`.

However, when people use Netlify CMS and don't specify the branch name to push content in the config file, Netlify CMS will automatically push to branch `master`.

Since this branch may not exist, users will get an error on the frontend: "**API_ERROR: Branch not found**". 


**Test plan**
1. In the `admin/config.yml` file, specify a valid Github repo path to push content to.
2. Run `yarn start`.
3. Go to Neltify CMS dashboard, create a new post and hit Publish. You shouldn't get an error. Checking back on the repo that you specified in the config, you should see the new post.



**Checklist**

Please add a `x` inside each checkbox:

- [x] Code is formatted via running `yarn format`.
- [ ] Tests are passing via running `yarn test`.
- [ ] The status checks are successful (continuous integration). Those can be seen below.
